src/codeu/chat/common/Message.java

```diff
@@ -55,7 +57,8 @@ public Message read(InputStream in) throws IOException {
   };
 
   public final Uuid id;
-  public final Uuid previous;
+  //TODO: got rid of final privacy setting, unsure if this is the right solution
```

Comment:
I would final because we wouldn't want anyone changing that value. That would be changing who the message was from, which is impossible.